kf/bpf_test.go

```diff
@@ -741,42 +770,229 @@ func Test_VerifyPinnedMapVanish(t *testing.T) {
 				Program: models.BPFProgram{
 					MapName: "",
 				},
+				hostConfig: &config.Config{
+					BpfMapDefaultPath: "/sys/fs/bpf",
+				},
 			},
 			wantErr: false,
 		},
 		{
 			name: "invalidProgType",
 			fields: fields{
 				Program: models.BPFProgram{
-					MapName:  "something",
+					MapName:  "tc/globals/something",
 					ProgType: models.TCType,
 				},
+				hostConfig: &config.Config{
+					BpfMapDefaultPath: "/sys/fs/bpf",
+				},
 			},
 			wantErr: false,
 		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			b := &BPF{
+				Program:      tt.fields.Program,
+				Cmd:          tt.fields.Cmd,
+				FilePath:     tt.fields.FilePath,
+				RestartCount: tt.fields.RestartCount,
+				hostConfig: &config.Config{
+					BpfMapDefaultPath: "/sys/fs/bpf",
+				},
+			}
+			err := b.VerifyPinnedMapVanish(true)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("VerifyPinnedMapVanish() error : %v", err)
+			}
+		})
+	}
+}
+
+func TestMapFullPath(t *testing.T) {
+	type fields struct {
+		Program      models.BPFProgram
+		Cmd          *exec.Cmd
+		FilePath     string
+		RestartCount int
+		hostConfig   *config.Config
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		result string
+	}{
+		{
+			name: "XdpTypeMapName",
+			fields: fields{
+				Program: models.BPFProgram{
+					ProgType: models.XDPType,
+					MapName:  "root_array",
+				},
+				Cmd:          nil,
+				RestartCount: 0,
+				hostConfig: &config.Config{
+					BpfMapDefaultPath: "/sys/fs/bpf",
+				},
+			},
+			result: "/sys/fs/bpf/root_array",
+		},
 		{
-			name: "invalidMapfile",
+			name: "TcTypeMapName",
+			fields: fields{
+				Program: models.BPFProgram{
+					ProgType: models.TCType,
+					MapName:  "tc/globals/tc_ingress_array",
+				},
+				Cmd:          nil,
+				RestartCount: 0,
+				hostConfig: &config.Config{
+					BpfMapDefaultPath: "/sys/fs/bpf",
+				},
+			},
+			result: "/sys/fs/bpf/tc/globals/tc_ingress_array",
+		},
+		{
+			name: "XdpTypeMapName_windows",
 			fields: fields{
 				Program: models.BPFProgram{
-					MapName:  "dummy",
 					ProgType: models.XDPType,
+					MapName:  "root_array",
+				},
+				Cmd:          nil,
+				RestartCount: 0,
+				hostConfig: &config.Config{
+					BpfMapDefaultPath: "",
 				},
 			},
-			wantErr: false,
+			result: "root_array",
+		},
+		{
+			name: "TcTypeMapName_windows",
+			fields: fields{
+				Program: models.BPFProgram{
+					ProgType: models.TCType,
+					MapName:  "tc\\globals\\tc_ingress_array",
```

Comment:
```suggestion
					MapName:  "tc/globals/tc_ingress_array",
```
these are not filesystem paths so no need to use \\